GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -478,6 +491,45 @@ private int safeCompare(Date date1 , Date date2 ) {
         log.debug("Finished data sort");
     }
 
+    //gather some statistics on the data collection
+    private void analyseData(final List<InstanceData> allData) {
+        totalInstances = allData.size();
+        
+        for (InstanceData instance : allData) {
+            SurveyInstanceDto sid = instance.surveyInstanceDto;
+            if (sid.getApprovedFlag() != null && sid.getApprovedFlag().equalsIgnoreCase("true")) {
+                approvedInstances++;
+            }
+            
+            Date cd = sid.getCollectionDate();
+            if (firstSubmission == null || cd.before(firstSubmission)) {
+                firstSubmission = cd;
+            }
+            if (lastSubmission == null || cd.after(lastSubmission)) {
+                lastSubmission = cd;
+            }
+            String u = sid.getSubmitterName();
+            if (instancesByUser.containsKey(u)) {
+                instancesByUser.put(u, instancesByUser.get(u) + 1);
+            } else {
+                instancesByUser.put(u, Integer.valueOf(1));
+            }
+            String dev = sid.getDeviceIdentifier();
+            if (instancesByDevice.containsKey(dev)) {
+                instancesByDevice.put(dev, instancesByDevice.get(dev) + 1);
+            } else {
+                instancesByDevice.put(dev, Integer.valueOf(1));
+            }
+            Long dur = sid.getSurveyalTime(); //seconds
```

Comment:
maybe dur should be called surveyalTimeInSeconds and then you can remove the comment

Reply:
Trying to get away from the confusing word "surveyal", but sure, I can lengthen it a bit.